### PR TITLE
e2e: add agent clusterCreate test

### PR DESCRIPTION
**Pull request by eranco74**
**What this PR does / why we need it**:
This PR adds a create cluster test for Agent platform

**Which issue(s) this PR fixes** *(optional, use `fixes #<issue_number>(, fixes #<issue_number>, ...)` format, where issue_number might be a GitHub issue, or a Jira story*:
Fixes # https://issues.redhat.com/browse/MGMT-9214

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [ ] This change includes docs. 
- [ ] This change includes unit tests.

### CODE REVIEW

**Comment by eranco74** [comment]
/hold
The test is currently failing to validate `image rollout`
For some reason the image update state (in the HC) is `Partial`

**Comment by sjenning** [comment]
```
=== CONT  TestAgentCreateCluster
    fixture.go:78: failed to create cluster, tearing down: failed to get agent namespace: resource name may not be empty
```

**Comment by eranco74** [comment]
> ```
> === CONT  TestAgentCreateCluster
>     fixture.go:78: failed to create cluster, tearing down: failed to get agent namespace: resource name may not be empty
> ```

How did you run the test?
Should be something like this:
`bin/test-e2e test.v -test.timeout 0 --e2e.agent-namespace hypershiftnodes  -e2e.base-domain redhat.com --e2e.control-plane-operator-image=quay.io/hypershift/hypershift-operator:latest --e2e.pull-secret-file ./ps -test.run='^TestAgent.*'`

Note that you should specify --e2e.agent-namespace that exists on the management cluster e.g https://github.com/openshift/release/pull/27591/files#diff-8de98e1c6b45dfcd057dd3229c06140b0cbc08dd4d9b54629026fe849c7dfaeeR367

Also note that the test requires a setup that creates the resources to be consumed by the agent platform (InfraEnv, Discovery ISO, BM / virual hosts running the iso and registered to the cluster) implemented [here](https://github.com/openshift/assisted-test-infra) and can be created using this command:

```
# Setup minikube as management cluster
ENABLE_KUBE_API=true make run
# create InfraEnv download ISO, boot 3 agent machines and approve the agent CR 
SPOKE_NAMESPACE=hypershiftnodes TEST_TEARDOWN=false CLUSTER_NAME=hypershift ENABLE_KUBE_API=true make test TEST=./src/tests/test_kube_api.py TEST_FUNC=test_prepare_late_binding_kube_api_ipv4_workers KUBECONFIG=$HOME/.kube/config
```

File: test/e2e/create_cluster_test.go [hunk 1]
```diff
@@ -154,3 +154,51 @@ func TestNoneCreateCluster(t *testing.T) {
...
+
+	clusterOpts := globalOpts.DefaultClusterOptions()
+	clusterOpts.Name = "hypershift"
+	hostedCluster := e2eutil.CreateCluster(t, ctx, client, &clusterOpts, hyperv1.AgentPlatform, globalOpts.ArtifactDir)
+
+	// Get the newly created nodepool
+	nodepool := &hyperv1.NodePool{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: hostedCluster.Namespace,
+			Name:      hostedCluster.Name,
```
Comment (enxebre) [inline test/e2e/create_cluster_test.go:180]: why does this test have to be different than the existing TestCreateCluster?

Comment (eranco74) [reply test/e2e/create_cluster_test.go:180]: I guess I can remove the agent specific validations and just use the existing TestCreateCluster

**Pull request closed without merging**